TicketUtil.py

```diff
@@ -119,6 +120,207 @@ def close_requests_session(self):
+        if self.ticket_id:
+            ticket_url = "{0}/show_bug.cgi?id={1}".format(self.base_url, self.ticket_id)
+
+        return ticket_url
+
+
+    def create_ticket_parameters(self, options_dict):
+        """
+        Creates the payload for the POST request when creating a Bugzilla ticket.
+
+        Example:
+        options_dict = {"product" : "TestProduct",
+                        "component" : "TestComponent",
+                        "version" : "unspecified",
+                        "summary" : "'This is a test bug - please disregard",
+                        "alias" : "SomeAlias",
+                        "op_sys" : "All",
+                        "priority" : "P1",
+                        "rep_platform" : "All"}
+        :param options_dict: Key: Value pairs for updating fields in Bugzilla.
+        :return: params: A dictionary to pass in to the POST request containing ticket details.
+        """
+        # Create our parameters for creating the ticket.
+        params = {'fields': {}}
+        params['fields']['project'] = {'key': self.project_key}
+
+        params = {"product": str(self.project_key)}
+
+        # Iterate through our options and add them to the params dict.
+        params.update(options_dict)
+        return params
+
+    def create_requests_session(self):
+        """
+        Returns to the super class if the authentication method is kerberos.
+        Creates a Requests Session and authenticates to base API URL with authentication other then kerberos.
+        We're using a Session to persist cookies across all requests made from the Session instance.
+        :return s: Requests Session.
+        """
+        if self.auth != 'rest':
+            return super()
+
+        try:
+            s = requests.Session()
+            r = s.get(self.auth_url, params=self.credentials, verify=False)
+            r.raise_for_status()
+            resp = r.json()
+            self.token = resp['token']
+            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+            logging.info("Successfully authenticated to {0} with token: {1}".format(self.ticketing_tool, self.token))
+            return s
+
+        # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+        except requests.RequestException as e:
+            logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
+            logging.error(e.args[0])
+
+
+    def create_ticket(self, params):
+        """
+        Tries to create the ticket through the ticketing tool's API.
+        Retrieves the ticket_id and creates the ticket_url.
+        :param params: The payload to send in the POST request.
+        :return:
+        """
+        # Attempt to create ticket.
+        try:
+
+            if self.token:
+                params['token'] = self.token
+
+            r = self.s.post(self.rest_url, params=params)
```

Comment:
One more quick comment: Does this POST request work with json=params instead of params=params?

I see you're using json=params when editing fields and adding comments, but params=params when creating the ticket. It's not a huge deal if they both work, but you could change this line to json=params (if it works), just to keep the code consistent.